mlflow/models/resources.py

```diff
@@ -77,6 +77,24 @@ class DatabricksResource(Resource, ABC):
     def target_uri(self) -> str:
         return "databricks"
 
+    @property
+    def type(self) -> ResourceType:
+        raise NotImplementedError("Subclasses must implement the 'type' property.")
+
+    def __init__(self, name: str, on_behalf_of_user: Optional[bool] = None):
```

Comment:
```suggestion
    def __init__(self, name: str, on_behalf_of_user: bool = False):
```

Does `None` has special meaning here? Otherwise I would set a default value to be false for simplicity.

Reply:
Added none here to not cause any unneccessary changes for existing behaviour. When a yaml file is read and converted to a dictionary, currently there is no `on_behalf_of_user` key. In order to not change this behaviour, I introduced a None as well

Reply:
```
Added none here to not cause any unneccessary changes for existing behaviour.
```
Got it. So for example, we are not 100% confident that setting `on_behalf_of_user` to `False` has exact same behavior as absence of the key - am I understanding it right?

If that case, this sounds like change management rather than backward compatibility. In that case, can we file a clean-up TODO JIRA + comment to remove `Optional` after releasing the feature? This is user-facing and we want to remove confusion.